tiledb/sm/filter/test/unit_run_filter_pipeline.cc

```diff
@@ -0,0 +1,1546 @@
+/**
+ * @file unit-run-filter-pipeline.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This set of unit tests checks running the filter pipeline forward and
+ * backward. Most tests here use simplified filters that perform basic
+ * operations such as adding 1 to each value in the original data so that the
+ * filtered data itself can be checked after running the filter pipeline
+ * forward.
+ */
+
+#include <random>
+
+#include <test/support/tdb_catch.h>
+
+#include "../bit_width_reduction_filter.h"
+#include "../bitshuffle_filter.h"
+#include "../byteshuffle_filter.h"
+#include "../checksum_md5_filter.h"
+#include "../checksum_sha256_filter.h"
+#include "../compression_filter.h"
+#include "../encryption_aes256gcm_filter.h"
+#include "../filter.h"
+#include "../filter_pipeline.h"
+#include "../positive_delta_filter.h"
+#include "add_1_in_place_filter.h"
+#include "add_1_including_metadata_filter.h"
+#include "add_1_out_of_place_filter.h"
+#include "add_n_in_place_filter.h"
+#include "pseudo_checksum_filter.h"
+#include "tiledb/sm/crypto/encryption_key.h"
+#include "tiledb/sm/enums/compressor.h"
+#include "tiledb/sm/enums/datatype.h"
+#include "tiledb/sm/enums/encryption_type.h"
+#include "tiledb/sm/enums/filter_option.h"
+#include "tiledb/sm/enums/filter_type.h"
+#include "tiledb/sm/tile/tile.h"
+
+using namespace tiledb::sm;
+
+// A dummy `Stats` instance.
+static tiledb::sm::stats::Stats dummy_stats("test");
+
+WriterTile make_increasing_tile(const uint64_t nelts) {
```

Comment:
These are getting duplicated here... Do you have any plan to remove this added duplication in further moves?

Reply:
They are going to eventually be removed/replaced in the new unit test directory. The remaining tests will also eventually be moved over.